Sources/JWTKit/JWTHeader.swift

```diff
@@ -1,20 +1,126 @@
 /// The header (details) used for signing and processing the JWT.
-package struct JWTHeader: Codable, Sendable {
+public struct JWTHeader: Codable, Sendable {
     /// The algorithm used with the signing.
-    package var alg: String?
+    public var alg: String?
 
     /// The Signature's Content Type.
-    package var typ: String?
+    public var typ: String?
 
     /// The Payload's Content Type.
-    var cty: String?
-
-    /// Critical fields.
-    var crit: [String]?
+    public var cty: String?
 
     /// The JWT key identifier.
-    var kid: JWKIdentifier?
+    public var kid: JWKIdentifier?
 
     /// The x5c certificate chain.
-    package var x5c: [String]?
+    public var x5c: [String]?
+
+    /// Custom fields.
+    public var customFields: [String: JWTHeaderField]?
```

Comment:
There's no reason for this to be `Optional` - an empty dictionary is identical in meaning to a `nil` one and requires less code acrobatics.